products/bash/mq/Readme.md

```diff
@@ -0,0 +1,218 @@
+    SET AUTHREC PROFILE('CrumpledOut') PRINCIPAL('mqm') OBJTYPE(QUEUE) AUTHADD(BROWSE,GET,INQ,PUT) \n\
+```
+
+- Setup certs,keys,keystores,truststores for MQ
+    
+    We have used this [products\bash\mq\createcerts\generate-test-cert.sh] to create the certs.
+    When creating application.kdb we need to add the `aceclient` label so that ACE Integration Servers can recognize it
+
+    I used the following command to add the label:
+
+    `runmqckm -cert -import -file application.p12 -pw password -type pkcs12 -target application.kdb -target_pw password -target_type cms -label "1" -new_label aceclient`
+
+    Note: runmqckm utility is available as a part of IBM MQ Client package and needs to be installed separately.
+
+- Setup CR and secrets for MQ:
+  
+    For MQ to utilize the certs that are part of the trust and keystore we create a secret called `mqcert` with the server key, server cert and application cert.
+
+    ```
+    QM_KEY=$(cat $CURRENT_DIR/mq/createcerts/server.key | base64 -w0)
+    QM_CERT=$(cat $CURRENT_DIR/mq/createcerts/server.crt | base64 -w0)
+    APP_CERT=$(cat $CURRENT_DIR/mq/createcerts/application.crt | base64 -w0)
+
+    kind: Secret
+    apiVersion: v1
+    metadata:
+      name: mqcert
+      namespace: $namespace
+    data:
+      tls.key: $QM_KEY
+      tls.crt: $QM_CERT
+      app.crt: $APP_CERT
+    type: Opaque
+    ```
+
+    For MQ to use this secret and to enable auth we use a modified CR of MQ:
+
+    ```
+    apiVersion: mq.ibm.com/v1beta1
+    kind: QueueManager
+    metadata:
+    name: ${release_name}
+    namespace: ${namespace}
+    spec:
+    license:
+        accept: true
+        license: L-RJON-BN7PN3
+        use: NonProduction
+    pki:
+        keys:
+        - name: default
+            secret:
+            items:
+                - tls.key
+                - tls.crt
+            secretName: mqcert
+        trust:
+        - name: app
+            secret:
+            items:
+                - app.crt
+            secretName: mqcert
+    queueManager:
+        image: ${image_name}
+        imagePullPolicy: Always
+        name: ${qm_name}
+        storage:
+        queueManager:
+            type: ephemeral
+        ini:
+        - configMap:
+            items:
+                - example.ini
+            name: mtlsmqsc
+    template:
+        pod:
+        containers:
+            - env:
+                - name: MQS_PERMIT_UNKNOWN_ID
+                value: 'true'
+            name: qmgr
+    version: 9.2.0.0-r1
+    web:
+        enabled: true
+    tracing:
+        enabled: ${tracing_enabled}
+        namespace: ${tracing_namespace}
+    ```
+
+## Setting up ACE to use TLS
+
+For ACE to use the TLS configuration we need provide the following `ace-configuration` to ACE instegration server CR:
+
+```
+serverconf:
+serverConfVersion: 1
+BrokerRegistry:
+mqKeyRepository: /home/aceuser/keystores/application
```

Comment:
What are these lines? Should they be part of a CR?

Reply:
thats a sample serverconf.yaml

Reply:
fixed the indentation for it